### PR TITLE
Search Facet Component - Part 1

### PR DESCRIPTION
**Pull request by thostetler**
This includes the bulk of the updates for the search facets, still working on the query updater part.

Search Facet Example

### CODE REVIEW

**Review by shinyichen — APPROVED** [review]
Cool!

Just a few questions.

File: package.json [hunk 1]
```diff
@@ -21,15 +21,17 @@
...
+    "@dnd-kit/modifiers": "^6.0.0",
+    "@dnd-kit/sortable": "^7.0.1",
     "@emotion/react": "^11.7.0",
     "@emotion/styled": "^11.6.0",
     "@headlessui/react": "^1.1.1",
     "@heroicons/react": "^1.0.1",
-    "@nivo/bar": "^0.78.0",
-    "@nivo/core": "^0.78.0",
-    "@nivo/line": "^0.78.0",
+    "@nivo/bar": "^0.79.1",
```
[the remark this answers is not in the document]
Comment (thostetler) [reply package.json:31]: Yep I'll pull in whatever version is in master

File: src/components/SearchFacet/useGetFacetTreeData.ts [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+      dispatch({ type: 'MORE' });
+    }
+  }, [state.canLoadMore]);
+
+  // update internal state with new data, when something changes
+  useEffect(() => {
+    if (data) {
+      dispatch({ type: 'DATA', payload: parseFacetDataIntoTuples(field, data, property, filter) });
+    }
+  }, [data, field, property]);
```
[the remark this answers is not in the document]
Comment (thostetler) [reply src/components/SearchFacet/useGetFacetTreeData.ts:108]: Yep, you're right --- I'll fix

File: src/components/__stories__/YearsGraphPane.stories.tsx [hunk 3]
```diff
@@ -13,4 +12,4 @@ const Template: Story<IYearsGraphPaneProps> = (args) => <YearsGraphPane {...args
 
 export const Default = Template.bind({});
 
-Default.args = { data: facetFoundFieldsData };
```
[the remark this answers is not in the document]
Comment (thostetler) [reply src/components/__stories__/YearsGraphPane.stories.tsx:16]: Yeah it was giving a typing error, I don't remember what, but I'll confirm

**Pull request merged**